libr/anal/p/anal_pickle.c

```diff
@@ -364,6 +364,288 @@ static int analop(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len, RAn
 	return op->size;
 }
 
+static inline bool write_num_sz(ut64 n, int byte_sz, ut8 *outbuf, int outsz) {
+	int bits = r_num_to_bits (NULL, n);
+	// TODO: signedness prbly wrong...
+	if (bits > byte_sz * 8) {
+		R_LOG_ERROR ("Arg 0x" PFMT64x " more then %d bits", n, bits);
+		false;
+	}
+	switch (byte_sz) {
+	case 1:
+		r_write_ble8 (outbuf, (ut8)(n & UT8_MAX));
+		break;
+	case 2:
+		r_write_ble16 (outbuf, (ut16)(n & UT16_MAX), false);
+		break;
+	case 4:
+		r_write_ble32 (outbuf, (ut32)(n & UT32_MAX), false);
+		break;
+	case 8:
+		r_write_ble64 (outbuf, n, false);
+		break;
+	default:
+		return false;
+	}
+	return true;
+}
+
+static inline int assemble_int(const char *str, int byte_sz, ut8 *outbuf, int outsz) {
+	if (outsz < byte_sz) {
+		return -2;
+	}
+	RNum *num = r_num_new (NULL, NULL, NULL);
+	if (num) {
+		ut64 n = r_num_math (num, str);
+		r_num_free (num);
+		if (write_num_sz (n, byte_sz, outbuf, outsz)) {
+			return byte_sz;
+		}
+	}
+	return 0;
+}
+
+static inline int assemble_float(const char *str, ut8 *outbuf, int outsz) {
+	if (outsz < sizeof (double)) {
+		return -2;
+	}
+	RNum *num = r_num_new (NULL, NULL, NULL);
+	if (num) {
+		*((double *)outbuf) = r_num_get_float (num, str);
+		r_mem_swap (outbuf, sizeof (double));
+		r_num_free (num);
+		return sizeof (double);
+	}
+	return 0;
+}
+
+static inline int assemble_cnt_str(const char *str, int byte_sz, ut8 *outbuf, int outsz) {
+	int wlen = -2;
+	char *write_str = strdup (str);
+	if (write_str) {
+		int len = r_str_unescape (write_str);
+		if (len > 0 && len + byte_sz <= outsz && write_num_sz (len, byte_sz, outbuf, outsz)) {
+			wlen = len + byte_sz;
+			memcpy (outbuf + byte_sz, write_str, len);
+		}
+		free (write_str);
+	}
+	return wlen;
+}
+
+static inline int write_op(char *opstr, ut8 *outbuf) {
+	struct opmap {const char *name; char op;};
```

Comment:
Make that global and static const

Reply:
Good call, I am surprised how much of a speed up that produced.